CheckBytes<DefaultValidator<'_>> is not implemented

**Describe the bug**
During the use of this crate in the implementation of https://github.com/dusk-network/piecrust/issues/166, the following compilation error occurs.

```text
error[E0277]: the trait bound `ArchivedTree<store::hash::Hash, 32, 4>: bytecheck::CheckBytes<DefaultValidator<'_>>` is not satisfied
   --> piecrust/src/store.rs:296:16
    |
296 |     let tree = rkyv::from_bytes(&tree_bytes).map_err(|err| {
    |                ^^^^^^^^^^^^^^^^ the trait `bytecheck::CheckBytes<DefaultValidator<'_>>` is not implemented for `ArchivedTree<store::hash::Hash, 32, 4>`

```

**Expected behaviour**
It should be possible to validate the structure on disk.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,7 +12,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Derive `Debug`, `Clone`, `PartialEq`, `Eq`, and `Hash` for `Tree` and `Opening` [#13]
 - Add `blake3` feature, implementing `Aggregate` for `blake3::Hash` [#11]
 
+### Fixed
+
+- Fix `CheckBytes` derivation in `Node` [#15]
+
 <!-- ISSUES -->
+[#15]: https://github.com/dusk-network/merkle/issues/15
 [#13]: https://github.com/dusk-network/merkle/issues/13
 [#11]: https://github.com/dusk-network/merkle/issues/11
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,7 +12,7 @@ license = "MPL-2.0"
 
 [dependencies]
 rkyv = { version = "0.7", optional = true, default-features = false }
-bytecheck = { version = "0.7", optional = true, default-features = false }
+bytecheck = { version = "0.6", optional = true, default-features = false }
 blake3 = { version = "1", optional = true, default-features = false }
 
 [dev-dependencies]
@@ -24,6 +24,7 @@ size_16 = ["rkyv/size_16"]
 size_32 = ["rkyv/size_32"]
 size_64 = ["rkyv/size_64"]
 rkyv-impl = [
+    "rkyv/validation",
     "rkyv/alloc",
     "rkyv",
     "bytecheck"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -23,9 +23,12 @@ use alloc::collections::BTreeSet;
 use core::mem;
 
 #[cfg(feature = "rkyv-impl")]
-use bytecheck::CheckBytes;
+use bytecheck::{CheckBytes, Error as BytecheckError};
 #[cfg(feature = "rkyv-impl")]
-use rkyv::{ser::Serializer, Archive, Deserialize, Serialize};
+use rkyv::{
+    ser::Serializer, validation::ArchiveContext, Archive, Deserialize,
+    Fallible, Serialize,
+};
 
 pub use aggregate::*;
 pub use opening::*;
@@ -35,12 +38,18 @@ pub use opening::*;
     feature = "rkyv-impl",
     derive(Archive, Serialize, Deserialize),
     archive(bound(serialize = "__S: Serializer")),
-    archive_attr(derive(CheckBytes), doc(hidden))
+    archive_attr(
+        derive(CheckBytes),
+        doc(hidden),
+        check_bytes(
+            bound = "__C: ArchiveContext, <__C as Fallible>::Error: BytecheckError"
+        )
+    )
 )]
 #[doc(hidden)]
 pub struct Node<T, const H: usize, const A: usize> {
     item: T,
-    #[cfg_attr(feature = "rkyv-impl", omit_bounds)]
+    #[cfg_attr(feature = "rkyv-impl", omit_bounds, archive_attr(omit_bounds))]
     children: [Option<Box<Node<T, H, A>>>; A],
 }
 

